FC-Chess/FC-Chess.cpp

```diff
@@ -107,28 +105,271 @@ void FCC::Chess::checkMovesHorizontal()
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i += 3, j -= 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i -= 3, j += 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i -= 3, j -= 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+}
+
+void FCC::Chess::checkMovesKing()
+{
+    // creating variables for x and y
+    uint8_t i, j;
+
+    // checking possible moves
+    i = selX - 1, j = selY + 1;
+    for (; i <= selX + 1; i++)
+    {
+        if (checkIsPosInBounds(&i, &j))
+        {
+            if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+        }
     }
+
+    i = selX - 1, j = selY - 1;
+    for (; i <= selX + 1; i++)
+    {
+        if (checkIsPosInBounds(&i, &j))
+        {
+            if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+        }
+    }
+
+    i = selX - 1, j = selY;
+    if (checkIsPosInBounds(&i, &j))
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX + 1, j = selY;
+    if (checkIsPosInBounds(&i, &j))
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+}
+
+void FCC::Chess::checkMovesPawn()
+{
+    // black pieces
+    if (posColor[selX][selY] == FCC::PiecesColors::BLACK)
+    {
+        // creating variables for x and y
+        uint8_t i, j;
+
+        // checking possible moves
+        i = selX - 1, j = selY - 1;
+        for (; i <= selX + 1;)
+        {
+            if (checkIsPosInBounds(&i, &j))
+            {
+                if (checkIsEnemyOnField(&i, &j)) posPossible[i][j] = true;
+            }
+            i += 2;
+        }
+
+        i = selX, j = selY - 1;
+        if (checkIsPosInBounds(&i, &j))
+        {
+            if (checkIsPositionClear(&i, &j)) posPossible[i][j] = true;
+        }
+    }
+
+    // white pieces
+    if (posColor[selX][selY] == FCC::PiecesColors::WHITE)
+    {
+        // creating variables for x and y
+        uint8_t i, j;
+
+        // checking possible moves
+        i = selX - 1, j = selY + 1;
+        for (; i <= selX + 1;)
+        {
+            if (checkIsPosInBounds(&i, &j))
+            {
+                if (checkIsEnemyOnField(&i, &j)) posPossible[i][j] = true;
+            }
+            i += 2;
+        }
+
+        i = selX, j = selY + 1;
+        if (checkIsPosInBounds(&i, &j))
+        {
+            if (checkIsPositionClear(&i, &j)) posPossible[i][j] = true;
+        }
+    }
+}
+
+bool FCC::Chess::checkIsPawnChangingToQueen()
+{
+    //black pieces
+    if (posColor[selX][selY] == FCC::PiecesColors::BLACK)
+    {
+        if (selY == 0) return true;
+        return false;
```

Comment:
not needed when `else if` is used for the next if statement since it will fall through and then the `return false` on the end of the function will be used

Reply:
Applied that. Thanks.